kubernetes-addons/memory-bandwidth-exporter/cmd/main.go

```diff
@@ -0,0 +1,239 @@
+	level.Debug(h.logger).Log("msg", "collect query:", "filters", filters)
+
+	if len(filters) == 0 {
+		// No filters, use the prepared unfiltered handler.
+		h.unfilteredHandler.ServeHTTP(w, r)
+		return
+	}
+	// To serve filtered metrics, we create a filtering handler on the fly.
+	filteredHandler, err := h.innerHandler(filters...)
+	if err != nil {
+		level.Warn(h.logger).Log("msg", "Couldn't create filtered metrics handler:", "err", err)
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write([]byte(fmt.Sprintf("Couldn't create filtered metrics handler: %s", err)))
+		return
+	}
+	filteredHandler.ServeHTTP(w, r)
+}
+
+// innerHandler is used to create both the one unfiltered http.Handler to be
+// wrapped by the outer handler and also the filtered handlers created on the
+// fly. The former is accomplished by calling innerHandler without any arguments
+// (in which case it will log all the collectors enabled via command-line
+// flags).
+func (h *handler) innerHandler(filters ...string) (http.Handler, error) {
+	level.Info(h.logger).Log("msg", "filters", filters)
+	nc, err := collector.NewCollector(h.logger, *interval, filters...)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't create collector: %s", err)
+	}
+
+	// Only log the creation of an unfiltered handler, which should happen
+	// only once upon startup.
+	if len(filters) == 0 {
+		level.Info(h.logger).Log("msg", "Enabled collectors")
+		collectors := []string{}
+		for n := range nc.Collectors {
+			collectors = append(collectors, n)
+		}
+		sort.Strings(collectors)
+		for _, c := range collectors {
+			level.Info(h.logger).Log("collector", c)
+		}
+	}
+
+	r := prometheus.NewRegistry()
+	r.MustRegister(versioncollector.NewCollector("memory_bandwidth_exporter"))
+	if err := r.Register(nc); err != nil {
+		return nil, fmt.Errorf("couldn't register collector: %s", err)
+	}
+
+	var handler http.Handler
+	if h.includeExporterMetrics {
+		handler = promhttp.HandlerFor(
+			prometheus.Gatherers{h.exporterMetricsRegistry, r},
+			promhttp.HandlerOpts{
+				ErrorLog:            stdlog.New(log.NewStdlibAdapter(level.Error(h.logger)), "", 0),
+				ErrorHandling:       promhttp.ContinueOnError,
+				MaxRequestsInFlight: h.maxRequests,
+				Registry:            h.exporterMetricsRegistry,
+			},
+		)
+		// Note that we have to use h.exporterMetricsRegistry here to
+		// use the same promhttp metrics for all expositions.
+		handler = promhttp.InstrumentMetricHandler(
+			h.exporterMetricsRegistry, handler,
+		)
+	} else {
+		handler = promhttp.HandlerFor(
+			r,
+			promhttp.HandlerOpts{
+				ErrorLog:            stdlog.New(log.NewStdlibAdapter(level.Error(h.logger)), "", 0),
+				ErrorHandling:       promhttp.ContinueOnError,
+				MaxRequestsInFlight: h.maxRequests,
+			},
+		)
+	}
+
+	return handler, nil
+}
+
+func main() {
+	promlogConfig := &promlog.Config{}
+	flag.AddFlags(kingpin.CommandLine, promlogConfig)
+	kingpin.Version(version.Print("memory bandwidth exporter"))
```

Comment:
is there a MBE version number? this print statement mentions no number. Assign it something.

Reply:
Yes, the memory bandwidth exporter has been assigned a version number.